src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -258,28 +259,88 @@ def __len__(self) -> int:
         """
         return len(self._buffer)
 
-    @overload
-    def __getitem__(self, key: SupportsIndex) -> float:
-        """See the main __getitem__ method.
+    def window(
+        self,
+        start: int | datetime,
+        end: int | datetime,
+        fill_gaps: Optional[float] = None,
+        force_copy: bool = False,
+    ) -> ArrayLike:
```

Comment:
@llucax Agree to make most of the arguments keyword only.

With respect to `start` and `end`, both are fine. It depends which behavior we want if start > end. We could
* return an empty array,
* raise and exception,
* or return the array in reverse order.

To me the last option would only make sense if we support more array-like functionality, like real slices including step sizes. However, I would leave this for a later version since it complicates this quite a bit. We would have all functionality on the array that the method returns. Instead I would go with the first option. In fact I would stick as close to other array-like types as possible, but just support start and end. Therefore I don't think we need keywords for start and end as it feels natural to use it this way. As long as we clearly define this in the doc.